src/app/payments/send-lightning.ts

```diff
@@ -0,0 +1,653 @@
+          senderUsername,
+          recipientUsername,
+          pubkey: recipientPubkey,
+          paymentHash,
+        })
+        const { metadata, debitAccountAdditionalMetadata: additionalDebitMetadata } =
+          lnIntraLedgerMetadata
+
+        const recipientWalletDescriptor = paymentFlow.recipientWalletDescriptor()
+        if (recipientWalletDescriptor === undefined)
+          return new InvalidLightningPaymentFlowBuilderStateError()
+
+        return LedgerFacade.recordIntraledger({
+          description: paymentFlow.descriptionFromInvoice,
+          amount: {
+            btc: paymentFlow.btcPaymentAmount,
+            usd: paymentFlow.usdPaymentAmount,
+          },
+          senderWalletDescriptor: paymentFlow.senderWalletDescriptor(),
+          recipientWalletDescriptor,
+          metadata,
+          additionalDebitMetadata,
+        })
+      })
+      if (journal instanceof Error) return journal
+
+      const lndService = LndService()
+      if (lndService instanceof Error) return lndService
+
+      const deletedLnInvoice = await lndService.cancelInvoice({
+        pubkey: recipientPubkey,
+        paymentHash,
+      })
+      if (deletedLnInvoice instanceof Error) return deletedLnInvoice
+
+      const newWalletInvoice = await WalletInvoicesRepository().markAsPaid(paymentHash)
+      if (newWalletInvoice instanceof Error) return newWalletInvoice
+
+      const notificationsService = NotificationsService(logger)
+      if (recipientWalletCurrency === WalletCurrency.Btc) {
+        notificationsService.lnInvoiceBitcoinWalletPaid({
+          paymentHash,
+          recipientWalletId,
+          sats: paymentFlow.btcPaymentAmount.amount,
+          displayCurrencyPerSat: priceRatio.usdPerSat(),
+        })
+      } else {
+        notificationsService.lnInvoiceUsdWalletPaid({
+          paymentHash,
+          recipientWalletId,
+          cents: paymentFlow.usdPaymentAmount.amount,
+          displayCurrencyPerSat: priceRatio.usdPerSat(),
+        })
+      }
+
+      return PaymentSendStatus.Success
+    },
+  )
+}
+
+const executePaymentViaLn = async ({
+  decodedInvoice,
+  paymentFlow,
+  senderWallet,
+  logger,
+}: {
+  decodedInvoice: LnInvoice
+  paymentFlow: PaymentFlow<WalletCurrency, WalletCurrency>
+  senderWallet: Wallet
+  logger: Logger
+}): Promise<PaymentSendStatus | ApplicationError> => {
+  const priceRatio = PriceRatio({
+    usd: paymentFlow.usdPaymentAmount,
+    btc: paymentFlow.btcPaymentAmount,
+  })
+  if (priceRatio instanceof Error) return priceRatio
+
+  const limitCheck = await newCheckWithdrawalLimits({
+    amount: paymentFlow.usdPaymentAmount,
+    wallet: senderWallet,
+    priceRatio,
+  })
+  if (limitCheck instanceof Error) return limitCheck
+
+  const { paymentHash } = decodedInvoice
+
+  // - get cached route if exists
+  const { rawRoute, outgoingNodePubkey } = paymentFlow.routeDetails()
+
+  // - validate route amount?
+
+  // - amount + fee in display currency?
```

Comment:
What is the purpose of this comment? Is this blocking?

Reply:
I can remove it from here actually, but the question is repeated here: https://github.com/GaloyMoney/galoy/pull/1237#discussion_r855521054